### PR TITLE
feat(controller): add file system support for StorageAccessService

### PR DESCRIPTION
**Pull request by xuchuan**
## Modules
- [ ] UI
- [X] Controller
- [ ] Agent
- [ ] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [ ] run code format and lint check
- [X] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: server/controller/src/main/resources/application.yaml [hunk 1]
```diff
@@ -34,6 +34,7 @@ sw:
       job-template-path: ${SW_K8S_JOB_TEMPLATE_PATH:}
   storage:
     path-prefix: ${SW_STORAGE_PREFIX:StarWhale}
+    fs-root-dir: ${SW_STORAGE_FS_ROOT_DIR:.}
```
Comment (goldenxinxing) [inline server/controller/src/main/resources/application.yaml:37]: How about using a fixed directory like “/opt/starwhale/storage”?

File: server/storage-access-layer/pom.xml [hunk 2]
```diff
@@ -36,6 +36,21 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-engine</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.hamcrest</groupId>
+            <artifactId>hamcrest</artifactId>
+            <version>2.2</version>
```
Comment (goldenxinxing) [inline server/storage-access-layer/pom.xml:51]: this version can managed by parent pom at block of "dependencyManagement", and  just use it without write version at childern pom

File: server/storage-access-layer/pom.xml [hunk 3]
```diff
@@ -49,6 +64,20 @@
             </dependency>
         </dependencies>
     </dependencyManagement>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
```
Comment (goldenxinxing) [inline server/storage-access-layer/pom.xml:71]: this version can also managed by parent pom at "pluginManagement"

File: server/controller/src/main/resources/application.yaml (hunk 1 above)
Comment (xuchuan) [reply server/controller/src/main/resources/application.yaml]: Done

File: server/storage-access-layer/pom.xml (hunk 2 above)
Comment (xuchuan) [reply server/storage-access-layer/pom.xml]: Done

File: server/storage-access-layer/pom.xml (hunk 3 above)
Comment (xuchuan) [reply server/storage-access-layer/pom.xml:71]: Done

File: server/storage-access-layer/src/main/java/ai/starwhale/mlops/storage/fs/StorageAccessServiceFile.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+    @Override
+    public void put(String path, InputStream inputStream, long size) throws IOException {
+        var f = new File(this.rootDir, path);
+        //noinspection ResultOfMethodCallIgnored
+        f.getParentFile().mkdirs();
+        var temp = File.createTempFile("sw_tmp", null);
+        try (var out = new FileOutputStream(temp)) {
+            inputStream.transferTo(out);
+            out.close();
```
Comment (anda-ren) [inline server/storage-access-layer/src/main/java/ai/starwhale/mlops/storage/fs/StorageAccessServiceFile.java:76]: `The try-with-resources statement ensures that each resource is closed at the end of the statement. Any object that implements java.lang.AutoCloseable, which includes all objects which implement java.io.Closeable, can be used as a resource`
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Comment (goldenxinxing) [reply server/storage-access-layer/src/main/java/ai/starwhale/mlops/storage/fs/StorageAccessServiceFile.java:76]: yes，so `out.close` could remove from here

Comment (xuchuan) [reply server/storage-access-layer/src/main/java/ai/starwhale/mlops/storage/fs/StorageAccessServiceFile.java:76]: There is a move operation followed by. Although it is not a problem on Linux, I guess it is a good idea to close the file before any further operation.

**Review by goldenxinxing — APPROVED** [review]
LGTM

**Pull request merged**